src/utils/QCCalculator.cpp

```diff
@@ -131,1062 +118,63 @@ class TOPPQCCalculator :
-    qp.cvAcc = "QC:0000057";
-    qp.value = String((100 / exp.size()) * below_10k);
-    try
-    {
-      const ControlledVocabulary::CVTerm& term = cv.getTerm(qp.cvAcc);
-      qp.name = term.name; ///< Name
-    }
-    catch (...)
-    {
-      qp.name = "percentage of ric slumps"; ///< Name
-    }
-    qcmlfile.addRunQualityParameter(base_name, qp);
-
-    qp = QcMLFile::QualityParameter();
-    qp.id = base_name + "_ricjump"; ///< Identifier
-    qp.cvRef = "QC"; ///< cv reference
-    qp.cvAcc = "QC:0000059";
-    qp.value = String(jumps);
-    try
-    {
-      const ControlledVocabulary::CVTerm& term = cv.getTerm(qp.cvAcc);
-      qp.name = term.name; ///< Name
-    }
-    catch (...)
-    {
-      qp.name = "IS-1A"; ///< Name
-    }
-    qcmlfile.addRunQualityParameter(base_name, qp);
-
-    qp = QcMLFile::QualityParameter();
-    qp.id = base_name + "_ricdump"; ///< Identifier
-    qp.cvRef = "QC"; ///< cv reference
-    qp.cvAcc = "QC:0000060";
-    qp.value = String(drops);
-    try
-    {
-      const ControlledVocabulary::CVTerm& term = cv.getTerm(qp.cvAcc);
-      qp.name = term.name; ///< Name
-    }
-    catch (...)
-    {
-      qp.name = "IS-1B"; ///< Name
-    }
-    qcmlfile.addRunQualityParameter(base_name, qp);
-
-    //---injection times MSn
-    at = QcMLFile::Attachment();
-    at.cvRef = "QC"; ///< cv reference
-    at.cvAcc = "QC:0000018";
-    at.qualityRef = msaq_ref;
-    at.id = base_name + "_ms2inj"; ///< Identifier
-    try
-    {
-      const ControlledVocabulary::CVTerm& term = cv.getTerm(at.cvAcc);
-      at.name = term.name; ///< Name
-    }
-    catch (...)
+    // collect QC data and store according to output file extension
+    if (FileHandler::hasValidExtension (outputfile_name, FileTypes::QCML)) 
     {
-      at.name = "MS2 injection time"; ///< Name
-    }
-
-    at.colTypes.push_back("MS:1000894_[sec]");
-    at.colTypes.push_back("MS:1000927");
-    for (Size i = 0; i < exp.size(); ++i)
+      QcMLFile qcmlfile;
+      qcmlfile.collectQCData(inputfile_id, inputfile_feature,
+                    inputfile_consensus, inputfile_raw, remove_duplicate_features, cv, exp);
+      qcmlfile.store(outputfile_name);
+    } else if (FileHandler::hasValidExtension (outputfile_name, FileTypes::MZQC))
     {
-      if (exp[i].getMSLevel() > 1 && !exp[i].getAcquisitionInfo().empty())
-      {
-        for (Size j = 0; j < exp[i].getAcquisitionInfo().size(); ++j)
-        {
-          if (exp[i].getAcquisitionInfo()[j].metaValueExists("MS:1000927"))
-          {
-            std::vector<String> row;
-            row.push_back(String(exp[i].getRT()));
-            row.push_back(exp[i].getAcquisitionInfo()[j].getMetaValue("MS:1000927"));
-            at.tableRows.push_back(row);
-          }
-        }
-      }
-    }
-    if (!at.tableRows.empty())
+      MzQCFile mzqcfile;
+      mzqcfile.store(inputfile_raw, outputfile_name, exp, cv, contactName, contactAddress, description, label);
+    } else 
```

Comment:
if you restrict the output filename in the register parameter section you don't need the else

Reply:
true, that was not necessary